src/Pirouette/Transformations/Contextualize.hs

```diff
@@ -0,0 +1,81 @@
+module Pirouette.Transformations.Contextualize where
+
+import qualified Data.Text as T
+import qualified Data.Map as Map
+import Pirouette.Monad
+import Pirouette.Term.Syntax as S
+import Pirouette.Term.Syntax.SystemF as SystF
+
+-- | Make all the 'Name's in the term with empty uniques
+-- refer to those in the context.
+-- This is especially useful when working with lonely terms
+-- created by the quasiquoter, but which ultimately need
+-- to refer back to elements in a 'Program'.
```

Comment:
```suggestion
-- to refer back to elements in a 'Program'. In particular, the `nameUnique` part might need adjustments.
-- Say you called the quasiquoter with @[pir| \x : Integer -> fib x |]@, you'd want `fib` to refers to something 
-- from the context. The `fib` in the context will actually be called @Name "fib" (Just u)@, where `u` is a 
-- unique identifier but the `fib` that came from the quasiquoter will be @Name "fib" Nothing@, hence we
-- need to adjust the 'nameUnique' portions.
```

Reply:
It took me a long time to realize why we even need this; I'd suggest appending some information to the comment